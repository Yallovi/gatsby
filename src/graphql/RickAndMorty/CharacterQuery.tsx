import { gql } from "@apollo/client";

export const FETCH_PAGE_CHARACTER = gql`
  query Query($characterId: ID!) {
    res: character(id: $characterId) {
      gender
      image
      name
      species
      status
    }
  }
`;
