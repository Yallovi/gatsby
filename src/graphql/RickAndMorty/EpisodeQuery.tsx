import { gql } from "@apollo/client";

// export * from './rick-and-morty'

export const FETCH_EPISODE = gql`
  query Query {
    res: episodes {
      results {
        id
        name
        episode
        airDate: air_date
        characters {
          id
          image
          name
        }
      }
    }
  }
`;

// export const FETCH_PAGE_CHARACTER = gql`
//   query Query($characterId: ID!) {
//     res: character(id: $characterId) {
//       gender
//       image
//       name
//       species
//       status
//     }
//   }
// `;
