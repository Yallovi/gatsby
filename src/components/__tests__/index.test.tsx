import React from "react";
import { cleanup, render } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { MockedProvider } from "@apollo/client/testing";
import Inedx from "~pages/index";
import { FETCH_EPISODE } from "~graphql/RickAndMorty/EpisodeQuery";
import { resultsQuery } from "../queryTest/EpisodeQueryTest";

afterEach(cleanup);

const mocks = [
  {
    request: {
      query: FETCH_EPISODE,
    },
    result: resultsQuery,
  },
];

it("renders Episode", async () => {
  const { findByText, getByText } = render(
    <MockedProvider mocks={mocks} addTypename={false}>
      <Inedx />
    </MockedProvider>
  );

  expect(getByText("loading")).toBeInTheDocument();
  const episodeNumber = await findByText("S02E09");
  expect(episodeNumber).toBeInTheDocument();
  expect(resultsQuery.data.res.results.length).toBe(20);
});
