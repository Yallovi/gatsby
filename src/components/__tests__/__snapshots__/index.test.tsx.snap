// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
Array [
  <h1>
    404: Not Found
  </h1>,
  <p>
    You just hit a route that doesn't exist... the sadness.
  </p>,
]
`;
